src/data/mv_polynomial.lean

```diff
@@ -739,6 +750,23 @@ by rw [vars, degrees_C, multiset.to_finset_zero]
 @[simp] lemma vars_X (h : 0 ≠ (1 : α)) : (X n : mv_polynomial σ α).vars = {n} :=
 by rw [X, vars_monomial h.symm, finsupp.support_single_ne_zero zero_ne_one.symm]
 
+lemma mem_support_not_mem_vars_zero {f : mv_polynomial σ α} {x : σ →₀ ℕ} (H : x ∈ f.support) {v : σ} (h : v ∉ vars f) :
+  x v = 0 :=
+begin
+  rw [vars, multiset.mem_to_finset] at h,
+  rw ←not_mem_support_iff,
+  by_contradiction,
+  have : v ∈ degrees f,
+  { unfold degrees,
+    rw (show f.support = insert x f.support, from eq.symm $ finset.insert_eq_of_mem H),
+    rw finset.sup_insert,
+    simp only [multiset.mem_union, multiset.sup_eq_union],
+    left,
+    rw [←to_finset_to_multiset, multiset.mem_to_finset] at a,
+    exact a },
```

Comment:
Here you are referring to `a`, which is a name introduced by `by_contradiction`. It would be better to give it a name explicitly, up there. So `by_contradiction H`. On the other hand, there is also the `contrapose` tactic, that you might want to use.